### PR TITLE
Moving unit/utils/test_jinja.py to pytest.

### PR DESCRIPTION
**Pull request by garethgreenaway**
### What does this PR do?
Moving unit/utils/test_jinja.py to pytest.

### Merge requirements satisfied?
**[NOTICE] Bug fixes or features added to Salt require tests.**

- [ ] Docs
- [ ] Changelog - https://docs.saltproject.io/en/master/topics/development/changelog.html
- [ ] Tests written/updated

### Commits signed with GPG?
Yes

Please review [Salt's Contributing Guide](https://docs.saltproject.io/en/master/topics/development/contributing.html) for best practices.

See GitHub's [page on GPG signing](https://help.github.com/articles/signing-commits-using-gpg/) for more information about signing commits with GPG.

### CODE REVIEW

File: tests/pytests/unit/utils/jinja/test_custom_extensions.py [hunk 1]
```diff
@@ -0,0 +1,1088 @@
...
+    with pytest.raises(exceptions.TemplateNotFound):
+        env.from_string('{% import_text "does not exists" as doc %}').render()
+
+
+def test_profile():
+    env = Environment(extensions=[SerializerExtension])
+
+    source = (
+        "{%- profile as 'profile test' %}"
+        + "{% set var = 'val' %}"
```
Comment (s0undt3ch) [inline tests/pytests/unit/utils/jinja/test_custom_extensions.py:262]:
```suggestion
        "{% set var = 'val' %}"
```
The `+` is not really necessary, python will concatenate the strings, right?

File: tests/pytests/unit/utils/jinja/test_custom_extensions.py [hunk 2]
```diff
@@ -0,0 +1,1088 @@
...
+    assert rendered == "16777216"
+
+
+@pytest.mark.requires_network
+def test_http_query(minion_opts, local_salt):
+    """
+    Test the `http_query` Jinja filter.
+    """
+    urls = (
+        # These cannot be HTTPS urls since urllib2 chokes on those
```
Comment (s0undt3ch) [inline tests/pytests/unit/utils/jinja/test_custom_extensions.py:758]: I know these were copied over, migrated, but are we still using urllib2 in our code?

File: tests/pytests/unit/utils/jinja/test_custom_extensions.py [hunk 3]
```diff
@@ -0,0 +1,1088 @@
...
+        "http://saltproject.io",
+        "http://google.com",
+        "http://duckduckgo.com",
+    )
+    for backend in ("requests", "tornado", "urllib2"):
```
Comment (s0undt3ch) [inline tests/pytests/unit/utils/jinja/test_custom_extensions.py:763]: The backend should be a test parametrization

File: tests/pytests/unit/utils/jinja/test_custom_extensions.py [hunk 4]
```diff
@@ -0,0 +1,1088 @@
...
+    Test the `json_query` Jinja filter.
+    """
+    rendered = render_jinja_tmpl(
+        "{{ [1, 2, 3] | json_query('[1]')}}",
+        dict(opts=minion_opts, saltenv="test", salt=local_salt),
+    )
+    assert rendered == "2"
+
+
+# def test_print():
```
Comment (s0undt3ch) [inline tests/pytests/unit/utils/jinja/test_custom_extensions.py:1079]: What's this one about?

File: tests/pytests/unit/utils/jinja/test_dot_notation_looup.py [hunk 5]
```diff
@@ -0,0 +1,64 @@
...
+def test_shadowed_dict_method():
+    """
+    Check calling a stub function with a name that shadows a ``dict``
+    method name
+    """
+    tmpl_str = """Hello, {{ salt.mockgrains.get('id') }}."""
+
+    with patch.object(SaltCacheLoader, "file_client", Mock()):
+        ret = render(tmpl_str)
+    assert ret == "Hello, jerry."
```
Comment (s0undt3ch) [inline tests/pytests/unit/utils/jinja/test_dot_notation_looup.py:64]: We could probably create a helper class that is module scoped.
Setup minion config once, define a render method which does the cache loader patching.
That should save some repetitive code.
Ping me if you want more details about this.

File: tests/pytests/unit/utils/jinja/test_get_template.py [hunk 6]
```diff
@@ -0,0 +1,487 @@
...
+    assert response == "2002-12-25"
+
+    objects = (
+        datetime.datetime(2002, 12, 25, 12, 00, 00, 00),
+        "2002/12/25",
+        1040814000,
+        "1040814000",
+    )
+
+    for object in objects:
```
Comment (s0undt3ch) [inline tests/pytests/unit/utils/jinja/test_get_template.py:342]: Parametrize on objects?

**Review by s0undt3ch — COMMENTED** [review]
This is what I have so far

File: tests/pytests/unit/utils/jinja/test_custom_extensions.py (hunk 4 above)
Comment (garethgreenaway) [reply tests/pytests/unit/utils/jinja/test_custom_extensions.py:1079]: No idea. Was commented out in the original.  Deleting.

**Comment by Ch3LL** [comment]
ping @s0undt3ch are you done with your review?

**Review by s0undt3ch — APPROVED** [review]
Yup. Done.

**Pull request merged**